Tensor.lua

```diff
@@ -525,6 +525,25 @@ function Tensor.totable(tensor)
   return result
 end
 
+function Tensor.permute(tensor, ...)
+  local perm = {...}
+  local nDims = tensor:dim()
+  assert(#perm == nDims, 'Invalid permutation')
```

Comment:
Could you make this constraint `#perm <= nDims`, and update doc and unit tests to reflect this?